src/update/balrog.cpp

```diff
@@ -300,6 +300,8 @@ bool Balrog::processData(const QByteArray& data) {
               // will be aborted, but the signal emitted.
               if (request->statusCode() == 200) {
                 emit updateRecommended();
+              } else {
+                emit updateNotAvailable();
```

Comment:
If you see, `Balrog::processData` has many reasons to do not call this signal: it has many `return false`. Same for `VersionApi`. I would suggest a different approach.

1. a method to know the status of the updater:
```
Class Updater ... {
  bool recommendedOrRequired() const { return m_recommendedOrRequired; }
  private:
  m_recommendedOrRequired = false;
}
```

2. let's use signals to set that boolean:
```
Updater::Updater() {
  connect(this, &updateRecommended, [this] { m_recommendedOrRequired = true; });
  connect(this, &updateRequired, [this] { m_recommendedOrRequired = true; });
 ...
```

3. in `ReleaseMonitor` do:
```
connect(updater, &QObject::destroyed, [this, updater]() {
  if (!updater->recommendedOrRequired()) emit notAvailable(); });
```

Thoughts?

Reply:
Good point!